emacs/ocamlformat.el

```diff
@@ -189,22 +194,21 @@ function."
 ;; replace-buffer-contents is broken in emacs-26.1
 ;; try to detect broken implementation.
 ;; https://bugzilla.redhat.com/show_bug.cgi?id=1597251
-(setq ocamlformat--support-replace-buffer-contents
-  (if (fboundp 'replace-buffer-contents)
-      (let* ((a (generate-new-buffer "tmp"))
-	     (b (generate-new-buffer "tmp")))
-	(with-current-buffer a
-	  (erase-buffer)
-	  (insert "\u2666\nabc\n"))
-	(let ((ok (with-current-buffer b
-		    (erase-buffer)
-		    (insert "\u2666\naXbc\n")
-		    (replace-buffer-contents a)
-		    (string= (buffer-string) "\u2666\nabc\n"))))
-	  (kill-buffer a)
-	  (kill-buffer b)
-	  ok))
-    nil))
+(defconst ocamlformat--support-replace-buffer-contents
+  (and (fboundp 'replace-buffer-contents)
```

Comment:
What is this diff about ?

Reply:
Ah, this just replaces `(if cond expr nil)` with `(and cond expr)` since they are equivalent, and it sort of makes semantic sense to use an "and" here.  I can revert this if the diff is not desirable.